app/services/payment/spam_protection_service.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module Payment
+  class SpamProtectionService
+    def initialize(account, payment_status, user_session = nil)
```

Comment:
Not mandatory but we could skip the nil check that we have below by using here the _Null Object pattern_:
```suggestion
    def initialize(account, payment_status, user_session = UserSession.new)
```